### ISSUE TITLE
setClassToggle only supports single classes

### ISSUE THREAD

**Issue by sambaker** [janpaepke/ScrollMagic#313]
The documentation states that passing "class1 class2" to setClassToggle will work but it throws an error:

`Uncaught InvalidCharacterError: Failed to execute 'add' on 'DOMTokenList': The token provided ('blur-out blur1') contains HTML space characters, which are not valid in tokens.`

I'm using v2.0.5.

**Comment by janpaepke** [comment]
Sounds like a bug, that might stem from a time, when ScrollMagic was jQuery dependent.
I'll look into it.

**Comment by Alexand3r3** [comment]
I have the same problem..How I can fix it and add multiple css classes?

**Comment by janpaepke** [comment]
until it's fixed either use multiple clases or try using the jquery extension (think it should work).

**Comment by zzzbra** [comment]
Having this same problem. Have tried to use multiple scenes, but doing so has my DOM flailing madly, in this case adding the two classes I'm attempting to add with the two separate scenes and then quickly overwriting them. So, that doesn't seem to be a working solution.

**Comment by Master244** [comment]
what jquery extension? I really need this feature :)

**Comment by jonw100** [comment]
ditto Master244.

I'm having the same problem - Can you advise what is this jquery extension you mentioned?

And is this scheduled to be fixed at some point?

thanks

**Comment by Ihatetomatoes** [comment]
You are right, this is still an issue, but you can achieve a similar effect by using the GreenSock .set tween to add multiple classes this way.

Here is an example on CodePen.

http://codepen.io/ihatetomatoes/pen/9e18df235da9abb2766a61990094a368

Hope that helps.

**Comment by dnfesp** [comment]
As a workaround you can overwrite ScrollMagic._util.addClass/removeClass with the following patched versions at the begining of your scrollmagic code:

ScrollMagic._util.addClass = function _patchedAddClass(elem, classname) {
      if (classname) {
        if (elem.classList) {
          classname.split(' ').forEach(function _addCls(c) {
            elem.classList.add(c);
          });
        } else elem.className += ' ' + classname;
      }
    };
    ScrollMagic._util.removeClass = function _patchedRemoveClass(elem, classname) {
      if (classname) {
        if (elem.classList) {
          classname.split(' ').forEach(function _removeCls(c) {
            elem.classList.remove(c);
          });
        } else elem.className = elem.className.replace(new RegExp('(^|\\b)' + classname.split(' ').join('|') + '(\\b|$)', 'gi'), ' ');
      }
    };

After this you'll be able to use the *setClassToggle* method as you would expect with multiple classes.

Cheers

**Comment by Ihatetomatoes** [comment]
Great, thanks for the fix @dnfesp, here is an updated CodePen as a proof that it works. 

[http://codepen.io/ihatetomatoes/pen/c220082b709a2cbf4283e15df1839173](http://codepen.io/ihatetomatoes/pen/c220082b709a2cbf4283e15df1839173)

**Comment by webdevnerdstuff** [comment]
@dnfesp patch works.

Any idea when v2.0.6 is going to be released that has this fix?

**Comment by joshuapease** [comment]
I've worked around this issue by using Scene events. 

It also provides much more flexibility than the setClassToggle method. 

http://scrollmagic.io/docs/ScrollMagic.Scene.html#event:enter

Using jQuery in this example, but Vanilla JS would obviously work too. 

```
  var $element = $('#my-element');

  var controller = new ScrollMagic.Controller();

  var scene = new ScrollMagic.Scene({
        triggerElement: $element[0],
        triggerHook: 0.5,
        duration: '100%'
      })
      .addTo(controller);

    scene.on('enter', function(){
      $element.addClass('active-class');
    });

    scene.on('leave', function(){
      $element.removeClass('active-class');
    });
```

**Comment by slikNET** [comment]
Can anybody help me?
I call setClassToggle twice in one scene and works only in the last element

$links.each(function () {
                var section = '#section-' + $(this).data('menuanchor');
                var height = $(section).outerHeight();

                new ScrollMagic.Scene({
                    triggerElement: section,
                    duration: height
                })
                .setClassToggle('body', 'pp-viewing-'+$(this).data('menuanchor')) // this not working
                .setClassToggle(this, 'active') // this working
                .addTo(controller);
            });

**Comment by janpaepke** [comment]
Sorry for the decade-long delay.

This was a real bug in v2 -- `classList.add()` doesn't accept space-separated class strings; each class must be a separate argument. The fix was to split the string and apply classes individually, which was addressed in later v2 patches.

v3 has no `setClassToggle`. You handle class changes yourself in `enter`/`leave` event handlers, where you can use `classList.add('class1', 'class2')` (modern browsers support multiple arguments) or any other approach you prefer.

---
**ScrollMagic v3 is now available** -- a complete rewrite with an entirely new architecture and API. v2 will not receive further updates (2.0.9 is the final release).

👉 `npm install scrollmagic@next` -- [Docs](https://github.com/janpaepke/ScrollMagic/blob/main/README.md) · [Release notes](https://github.com/janpaepke/ScrollMagic/releases/tag/v3.0.0-beta.0) · [npm](https://www.npmjs.com/package/scrollmagic/v/3.0.0-beta.0)
